src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ChunkedEncodingWriteStream.cs

```diff
@@ -18,6 +18,26 @@ public ChunkedEncodingWriteStream(HttpConnection connection) : base(connection)
             {
             }
 
+            public override void Write(ReadOnlySpan<byte> buffer)
+            {
+                HttpConnection connection = GetConnectionOrThrow();
+                Debug.Assert(connection._currentRequest != null);
+
+                if (buffer.Length == 0)
+                {
+                    connection.Flush();
+                    return;
+                }
+
+                // Write chunk length in hex followed by \r\n
+                connection.WriteHexInt32Async(buffer.Length, async: false).GetAwaiter().GetResult();
+                connection.WriteTwoBytesAsync((byte)'\r', (byte)'\n', async: false).GetAwaiter().GetResult();
+
+                // Write chunk contents followed by \r\n
+                connection.Write(buffer);
+                connection.WriteTwoBytesAsync((byte)'\r', (byte)'\n', async: false).GetAwaiter().GetResult();
```

Comment:
These async methods here are pretty simple and can easily get sync counterparts.